staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go

```diff
@@ -120,13 +123,48 @@ func newGCPAuthProvider(_ string, gcpConfig map[string]string, persister restcli
 	if err != nil {
 		return nil, err
 	}
-	return &gcpAuthProvider{cts, persister}, nil
+	return &gcpAuthProvider{cts, persister, useDefaultTokenSource}, nil
+}
+
+type transport struct {
+	TokenSource oauth2.TokenSource
+	Base        http.RoundTripper
+	Handler     func(*http.Response)
+}
+
+func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	transport := oauth2.Transport{
+		Source: t.TokenSource,
+		Base:   t.Base,
+	}
+
+	res, err := transport.RoundTrip(req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	t.Handler(res)
+
+	return res, nil
 }
 
 func (g *gcpAuthProvider) WrapTransport(rt http.RoundTripper) http.RoundTripper {
-	return &oauth2.Transport{
-		Source: g.tokenSource,
-		Base:   rt,
+	handler := func(res *http.Response) {
+		switch res.StatusCode {
+		case 401:
+			if g.useDefaultTokenSource {
+				glog.V(4).Infof("The application default credentials that were supplied are invalid for the target cluster")
```

Comment:
I suggest removing the special casing for default token source and change this log to "The credentials that were supplied..."